src/array.jl

```diff
@@ -368,44 +416,31 @@ function copyto!(dest::CatArrOrSub{T, N}, dstart::Integer,
         throw(MissingException("cannot copy array with missing values to an array with element type $T"))
     end
 
-    newlevels, ordered = mergelevels(isordered(dest), dlevs, slevs)
-    if isordered(dest) && (length(newlevels) != length(dlevs))
-        # Uncomment this when removing deprecation
-        # throw(OrderedLevelsException(newlevels[findfirst(!in(Set(dlevs)), newlevels)],
-        #                              dlevs))
-        Base.depwarn("adding new levels to ordered CategoricalArray destination " *
-                     "will throw an error in the future", :copyto!)
-        ordered &= isordered(src) | (length(newlevels) == length(dlevs))
-    end
-    # Exception: empty pool marked as ordered if new value is ordered
-    if isempty(dlevs) && isordered(src)
-        ordered = true
-    end
-    if ordered != isordered(dest)
-        isa(dest, SubArray) && throw(ArgumentError("cannot set ordered=$ordered on dest SubArray as it would affect the parent. Found when trying to set levels to $newlevels."))
-        ordered!(dest, ordered)
-    end
+    destp = dest isa SubArray ? parent(dest) : dest
 
-    # Simple case: replace all values
-    if !isa(dest, SubArray) && dstart == dstart == 1 && n == length(dest) == length(src)
-        # Set index to reflect refs
-        levels!(dpool, T[]) # Needed in case src and dest share some levels
-        levels!(dpool, index(spool))
-
-        # Set final levels in their visible order
-        levels!(dpool, newlevels)
+    # For partial copy, need to recompute existing refs
+    # TODO: for performance, avoid ajusting refs which are going to be overwritten
+    updaterefs = isa(dest, SubArray) ||
+        !(dstart == dstart == 1 && n == length(dest) == length(src))
```

Comment:
why `dstart == dstart` here? Should it be `dstart == sstart`?

Reply:
For reference, after inspecting we determined the first condition isn't needed.